aten/src/ATen/native/cuda/TensorTopK.cu

```diff
@@ -190,14 +542,20 @@ void launch_gather_topk_kernel(
       static_cast<INDEX_T>(topKSlices),                                 \
       static_cast<INDEX_T>(topKInfo.strides[collapseTopKDim]),          \
       indicesInfo,                                                      \
-      static_cast<INDEX_T>(indicesInfo.strides[collapseIndicesDim]));   \
-  C10_CUDA_KERNEL_LAUNCH_CHECK();
+      static_cast<INDEX_T>(indicesInfo.strides[collapseIndicesDim]));
+
+#define RUN_MB(INDEX_T, DIM, DIR)               \
+  if (should_use_multiblock_per_slice) {        \
+    RUN_K(INDEX_T, DIM, DIR, mbtopk::launch);   \
+  } else {                                      \
+    RUN_K(INDEX_T, DIM, DIR, sbtopk::launch);   \
+  }
 
 #define RUN_DIR(INDEX_T, DIM)                   \
   if (largest) {                                \
-    RUN_K(INDEX_T, DIM, true);                  \
+    RUN_MB(INDEX_T, DIM, true);                 \
   } else {                                      \
-    RUN_K(INDEX_T, DIM, false);                 \
+    RUN_MB(INDEX_T, DIM, false);                \
   }
```

Comment:
Can we make `largest` an argument instead of a template parameter? This would also help reduce binary size.